ec/ecresource/extensionresource/read.go

```diff
@@ -60,40 +81,40 @@ func extensionNotFound(err error) bool {
 	return errors.As(err, &extensionNotFound)
 }
 
-func modelToState(d *schema.ResourceData, model *models.Extension) error {
-	if err := d.Set("name", model.Name); err != nil {
-		return err
+func modelToState(model *models.Extension, state *modelV0) {
+	if model.Name != nil {
+		state.Name = types.String{Value: *model.Name}
+	} else {
+		state.Name = types.String{Null: true}
 	}
 
-	if err := d.Set("version", model.Version); err != nil {
-		return err
+	if model.Version != nil {
+		state.Version = types.String{Value: *model.Version}
+	} else {
+		state.Version = types.String{Null: true}
 	}
 
-	if err := d.Set("extension_type", model.ExtensionType); err != nil {
-		return err
+	if model.ExtensionType != nil {
+		state.ExtensionType = types.String{Value: *model.ExtensionType}
+	} else {
+		state.ExtensionType = types.String{Null: true}
 	}
 
-	if err := d.Set("description", model.Description); err != nil {
-		return err
-	}
+	state.Description = types.String{Value: model.Description}
```

Comment:
Why do we treat these attributes differently? Why are we explicitly setting these values to null in this resource?

Reply:
@pascal-hofmann , can you please help with an answer?

Reply:
Pointer vs. non-pointer.